update backend setting functions to make them more intuitive

Sohob opened the pull request:
PR for this trello card: https://trello.com/c/j18DE7Jg/1483-renaming-backend-setting-functions-to-make-them-more-intuitive

Could elaborate more on the differences in the docs, but thought it would complicate things. Also thought about adding an example for unset_backend, but it wasn't required in the task.

Thanks! ✨

VedPatwardhan reviewed (changes requested):
Just a small change, let's add a line about `ivy.unset_backend` in the `backend_setting.rst` too, so that we'd let users know that they can also clear all set backends by using `ivy.unset_backend`. 
Rest looks great, thanks @Sohob 🙂

CatB1t reviewed (approved):
lgtm! let's get the note about `ivy.unset_backend` added to the docs as @VedPatwardhan mentioned, then we're good to merge.

Sohob commented:
I added a sentence for `unset_backend` and added to the `previous_backend` description to distinguish between them. Let me know if I should elaborate more.
Thank you for your reviews! 🌟

VedPatwardhan reviewed (approved):
lgtm! Feel free to merge the PR. Thanks @Sohob @CatB1t @nassimberrada 🙂
